app/models/manager_refresh/inventory_collection.rb

```diff
@@ -341,9 +342,10 @@ def initialize(model_class: nil, manager_ref: nil, association: nil, parent: nil
                    check_changed: nil, custom_manager_uuid: nil, custom_db_finder: nil, arel: nil, builder_params: {},
                    inventory_object_attributes: nil, unique_index_columns: nil, name: nil, saver_strategy: nil,
                    parent_inventory_collections: nil, manager_uuids: [], all_manager_uuids: nil, targeted_arel: nil,
-                   targeted: nil, manager_ref_allowed_nil: nil)
+                   targeted: nil, manager_ref_allowed_nil: nil, secondary_refs: {})
```

Comment:
@agrare any naming idea? Also would be nice to get the interface right, so it's easier to do to refactoring of the inside.

@agrare @cben
maybe if can be just indexes?

```
:indexes => {
  :name_label_index => {
    :attributes => [:name, :label],
    :unique => true
  },
   :other_index => {
    :attributes => [:other],
    :unique => false
  },
}
```

or maybe we should wrap index as an object right away?

```
:indexes => [
  InventoryCollection::Index.new([:name, :label], :unique => true),
  InventoryCollection::Index.new([:other], :unique => false, :name => "other_index"
]
```

(we could also extract the @data_index into InventoryCollection::Data)

and we might not need to name the indexes at all I guess, we can just check if the combination of attributes is there as an index, even with O(c) if we use the attribute list as an index?